lua/src/macro_cache.cc

```diff
@@ -182,37 +251,94 @@ void macro_cache::_process_instance(instance_broadcast const& in) {
  */
 void macro_cache::_process_host(neb::host const& h) {
   logging::debug(logging::medium)
-    << "macro_cache: process host --- "
-    << "id = " << h.host_id
-    << " ; name = " << h.host_name;
+    << "lua: processing host '" << h.host_name << "' of id " << h.host_id;
   _hosts[h.host_id] = h;
 }
 
+/**
+ *  Process a host group event.
+ *
+ *  @param hg  The event.
+ */
+void macro_cache::_process_host_group(neb::host_group const& hg) {
+  logging::debug(logging::medium)
+    << "lua: processing host group '" << hg.name << "' of id " << hg.id;
+  if (hg.enabled)
+    _host_groups[hg.id] = hg;
+  else
+    _host_groups.remove(hg.id);
+}
+
+/**
+ *  Process a host group member event.
+ *
+ *  @param hgm  The event.
+ */
+void macro_cache::_process_host_group_member(
+       neb::host_group_member const& hgm) {
+  logging::debug(logging::medium)
+    << "lua: processing host group member "
+    << " (group_name: '" << hgm.group_name << "', group_id: " << hgm.group_id
+    << ", host_id: " << hgm.host_id << ")";
+  if (hgm.enabled)
+    _host_group_members.insert(hgm.host_id, hgm);
+  else
+    _host_group_members.remove(hgm.host_id);
```

Comment:
Removal should only concern this specific membership (host / hostgroup pair), not all memberships of the host.